crates/context/config/src/client/env/config/query/privileges.rs

```diff
@@ -63,10 +66,62 @@ impl<'a> Method<Starknet> for PrivilegesRequest<'a> {
     const METHOD: &'static str = "privileges";
 
     fn encode(self) -> eyre::Result<Vec<u8>> {
-        todo!()
+        // Split context_id into high/low parts
+        let context_bytes = self.context_id.as_bytes();
+        let mid_point = context_bytes
+            .len()
+            .checked_div(2)
+            .ok_or_else(|| eyre::eyre!("Length should be even"))?;
+        let (high_bytes, low_bytes) = context_bytes.split_at(mid_point);
+
+        // Convert to Felts and then to bytes
+        let mut result = Vec::new();
+        result.extend_from_slice(&Felt::from_bytes_be_slice(high_bytes).to_bytes_be());
+        result.extend_from_slice(&Felt::from_bytes_be_slice(low_bytes).to_bytes_be());
```

Comment:
`FeltPair` -- and other cases, idk if there would be more at this point, but I won't point them out anymore to not clutter the review

you can maybe search for `checked_div`